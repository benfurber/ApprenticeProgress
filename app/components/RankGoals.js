import React, { Component } from "react";
import { FlatList, StyleSheet } from "react-native";
import { List, ListItem, Tab, TabHeading, Tabs, Text } from "native-base";

import { colours, standards, successScaleArray } from "styles";

class RankGoals extends Component {
  render() {
    const { goals } = this.props;

    return (
      <Tabs
        style={styles.tabs}
        tabBarUnderlineStyle={{ backgroundColor: colours.black }}
      >
        {this._renderAllTitles(goals)}
      </Tabs>
    );
  }

  _renderAllTitles(goals) {
    let allTitles = [];
    let score = 5;

    while (score > -1) {
      const titleSet = this._findTitles(goals, score);
      if (titleSet.length > 0) {
        allTitles.push(this._renderTitleSet(titleSet, score));
      }
      score -= 1;
    }

    return allTitles;
  }

  _findTitles(goals, score) {
    let titleSet = [];

    goals.forEach(goal => {
      if (goal.score === score) {
        titleSet.push({ id: goal.id, title: goal.title });
      }
    });

    return titleSet;
  }

  _renderTitleSet(titleSet, score) {
    return (
      <Tab
        key={score}
        heading={
          <TabHeading>
            <Text style={{ color: successScaleArray[score] }}>{score}s</Text>
          </TabHeading>
        }
        style={{ backgroundColor: "transparent" }}
      >
        <List style={styles.list}>
          <FlatList
            data={titleSet}
            renderItem={({ item }) => (
              <ListItem>
                <Text style={{ color: successScaleArray[score] }}>
                  - {item.title}
                </Text>
              </ListItem>
            )}
            keyExtractor={item => item.id.toString()}
          />
        </List>
      </Tab>
    );
  }
}

const styles = StyleSheet.create({
  tabs: {
    backgroundColor: colours.transparentWhite,
    height: 300,
    ...standards.borders,
  },
  list: {
    minHeight: 240,
  },
});

export { RankGoals };
