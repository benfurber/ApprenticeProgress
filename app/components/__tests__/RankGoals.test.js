import React from "react";
import { shallow } from "enzyme";

import { RankGoals } from "../RankGoals";

describe("RankGoals", () => {
  it("has a valid snapshot", () => {
    const goals = [
      {
        id: 1,
        title: "Low Score",
        score: 2,
      },
      {
        id: 2,
        title: "High Score",
        score: 4,
      },
    ];
    const component = shallow(<RankGoals goals={goals} />);

    expect(component).toMatchSnapshot();
  });
});
