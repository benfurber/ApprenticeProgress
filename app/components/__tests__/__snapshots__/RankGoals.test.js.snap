// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankGoals has a valid snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RankGoals
    goals={
      Array [
        Object {
          "id": 1,
          "score": 2,
          "title": "Low Score",
        },
        Object {
          "id": 2,
          "score": 4,
          "title": "High Score",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Tab)
          heading={
            <Styled(TabHeadingNB)>
              <Styled(Text)
                style={
                  Object {
                    "color": "#b5ab37",
                  }
                }
              >
                4
                s
              </Styled(Text)>
            </Styled(TabHeadingNB)>
          }
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(List)
            style={
              Object {
                "minHeight": 240,
              }
            }
          >
            <FlatList
              data={
                Array [
                  Object {
                    "id": 2,
                    "title": "High Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Styled(List)>
        </Styled(Tab)>,
        <Styled(Tab)
          heading={
            <Styled(TabHeadingNB)>
              <Styled(Text)
                style={
                  Object {
                    "color": "#d35f3c",
                  }
                }
              >
                2
                s
              </Styled(Text)>
            </Styled(TabHeadingNB)>
          }
          style={
            Object {
              "backgroundColor": "transparent",
            }
          }
        >
          <Styled(List)
            style={
              Object {
                "minHeight": 240,
              }
            }
          >
            <FlatList
              data={
                Array [
                  Object {
                    "id": 1,
                    "title": "Low Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Styled(List)>
        </Styled(Tab)>,
      ],
      "contentProps": Object {},
      "initialPage": 0,
      "locked": false,
      "onChangeTab": [Function],
      "onScroll": [Function],
      "page": -1,
      "prerenderingSiblingsNumber": 0,
      "scrollWithoutAnimation": false,
      "style": Object {
        "backgroundColor": "#FFFFFFCF",
        "borderBottomWidth": 2,
        "borderColor": "#5A6268",
        "borderTopWidth": 2,
        "height": 300,
      },
      "tabBarPosition": "top",
      "tabBarUnderlineStyle": Object {
        "backgroundColor": "#3D3E44",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "children": <Styled(List)
            style={
              Object {
                "minHeight": 240,
              }
            }
          >
            <FlatList
              data={
                Array [
                  Object {
                    "id": 2,
                    "title": "High Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Styled(List)>,
          "heading": <Styled(TabHeadingNB)>
            <Styled(Text)
              style={
                Object {
                  "color": "#b5ab37",
                }
              }
            >
              4
              s
            </Styled(Text)>
          </Styled(TabHeadingNB)>,
          "style": Object {
            "backgroundColor": "transparent",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FlatList
              data={
                Array [
                  Object {
                    "id": 2,
                    "title": "High Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
            "style": Object {
              "minHeight": 240,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [
                Object {
                  "id": 2,
                  "title": "High Score",
                },
              ],
              "disableVirtualization": false,
              "horizontal": false,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": <Styled(List)
            style={
              Object {
                "minHeight": 240,
              }
            }
          >
            <FlatList
              data={
                Array [
                  Object {
                    "id": 1,
                    "title": "Low Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Styled(List)>,
          "heading": <Styled(TabHeadingNB)>
            <Styled(Text)
              style={
                Object {
                  "color": "#d35f3c",
                }
              }
            >
              2
              s
            </Styled(Text)>
          </Styled(TabHeadingNB)>,
          "style": Object {
            "backgroundColor": "transparent",
          },
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FlatList
              data={
                Array [
                  Object {
                    "id": 1,
                    "title": "Low Score",
                  },
                ]
              }
              disableVirtualization={false}
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
            "style": Object {
              "minHeight": 240,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [
                Object {
                  "id": 1,
                  "title": "Low Score",
                },
              ],
              "disableVirtualization": false,
              "horizontal": false,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Tab)
            heading={
              <Styled(TabHeadingNB)>
                <Styled(Text)
                  style={
                    Object {
                      "color": "#b5ab37",
                    }
                  }
                >
                  4
                  s
                </Styled(Text)>
              </Styled(TabHeadingNB)>
            }
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <Styled(List)
              style={
                Object {
                  "minHeight": 240,
                }
              }
            >
              <FlatList
                data={
                  Array [
                    Object {
                      "id": 2,
                      "title": "High Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </Styled(List)>
          </Styled(Tab)>,
          <Styled(Tab)
            heading={
              <Styled(TabHeadingNB)>
                <Styled(Text)
                  style={
                    Object {
                      "color": "#d35f3c",
                    }
                  }
                >
                  2
                  s
                </Styled(Text)>
              </Styled(TabHeadingNB)>
            }
            style={
              Object {
                "backgroundColor": "transparent",
              }
            }
          >
            <Styled(List)
              style={
                Object {
                  "minHeight": 240,
                }
              }
            >
              <FlatList
                data={
                  Array [
                    Object {
                      "id": 1,
                      "title": "Low Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </Styled(List)>
          </Styled(Tab)>,
        ],
        "contentProps": Object {},
        "initialPage": 0,
        "locked": false,
        "onChangeTab": [Function],
        "onScroll": [Function],
        "page": -1,
        "prerenderingSiblingsNumber": 0,
        "scrollWithoutAnimation": false,
        "style": Object {
          "backgroundColor": "#FFFFFFCF",
          "borderBottomWidth": 2,
          "borderColor": "#5A6268",
          "borderTopWidth": 2,
          "height": 300,
        },
        "tabBarPosition": "top",
        "tabBarUnderlineStyle": Object {
          "backgroundColor": "#3D3E44",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "children": <Styled(List)
              style={
                Object {
                  "minHeight": 240,
                }
              }
            >
              <FlatList
                data={
                  Array [
                    Object {
                      "id": 2,
                      "title": "High Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </Styled(List)>,
            "heading": <Styled(TabHeadingNB)>
              <Styled(Text)
                style={
                  Object {
                    "color": "#b5ab37",
                  }
                }
              >
                4
                s
              </Styled(Text)>
            </Styled(TabHeadingNB)>,
            "style": Object {
              "backgroundColor": "transparent",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FlatList
                data={
                  Array [
                    Object {
                      "id": 2,
                      "title": "High Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />,
              "style": Object {
                "minHeight": 240,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "id": 2,
                    "title": "High Score",
                  },
                ],
                "disableVirtualization": false,
                "horizontal": false,
                "initialNumToRender": 10,
                "keyExtractor": [Function],
                "maxToRenderPerBatch": 10,
                "numColumns": 1,
                "onEndReachedThreshold": 2,
                "renderItem": [Function],
                "scrollEventThrottle": 50,
                "updateCellsBatchingPeriod": 50,
                "windowSize": 21,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": <Styled(List)
              style={
                Object {
                  "minHeight": 240,
                }
              }
            >
              <FlatList
                data={
                  Array [
                    Object {
                      "id": 1,
                      "title": "Low Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />
            </Styled(List)>,
            "heading": <Styled(TabHeadingNB)>
              <Styled(Text)
                style={
                  Object {
                    "color": "#d35f3c",
                  }
                }
              >
                2
                s
              </Styled(Text)>
            </Styled(TabHeadingNB)>,
            "style": Object {
              "backgroundColor": "transparent",
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <FlatList
                data={
                  Array [
                    Object {
                      "id": 1,
                      "title": "Low Score",
                    },
                  ]
                }
                disableVirtualization={false}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />,
              "style": Object {
                "minHeight": 240,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "id": 1,
                    "title": "Low Score",
                  },
                ],
                "disableVirtualization": false,
                "horizontal": false,
                "initialNumToRender": 10,
                "keyExtractor": [Function],
                "maxToRenderPerBatch": 10,
                "numColumns": 1,
                "onEndReachedThreshold": 2,
                "renderItem": [Function],
                "scrollEventThrottle": 50,
                "updateCellsBatchingPeriod": 50,
                "windowSize": 21,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
